src/main/java/Domain/Repositories/MemoryShopRepository.java

```diff
@@ -10,13 +10,16 @@
 public class MemoryShopRepository implements ShopRepositoryInterface {
     private Map<Integer, Shop> _shops;
     private int _shopIdCounter;
+    private int _productIdCounter;
 
     public MemoryShopRepository(List<Shop> shops) {
         _shops = new HashMap<>();
         _shopIdCounter = 0;
+        _productIdCounter = 0;
         for (Shop shop : shops) {
             _shops.put(shop.getShopId(), shop);
             _shopIdCounter++;
+            _productIdCounter += shop.getAmoutOfProductInShop();
```

Comment:
pls make sure it is not making a mistake when we get wrong ids for products when loading the data